crypto/asn1/a_time.c

```diff
@@ -589,3 +589,42 @@ int ASN1_TIME_compare(const ASN1_TIME *a, const ASN1_TIME *b)
         return -1;
     return 0;
 }
+
+/*
+ * tweak for Windows
+ */
+#ifdef WIN32
+# define timezone _timezone
+#endif
+
+time_t asn1_string_to_time_t(char *asn1_string)
+{
+    ASN1_TIME *timestamp_asn1 = NULL;
+    struct tm *timestamp_tm = NULL;
+    time_t timestamp_local;
+    time_t timestamp_utc;
+
+    timestamp_asn1 = ASN1_TIME_new();
+    if (!ASN1_TIME_set_string(timestamp_asn1, asn1_string))
+    {
+        ASN1_TIME_free(timestamp_asn1);
+        return -1;
+    }
+
+    timestamp_tm = malloc(sizeof(*timestamp_tm));
```

Comment:
Use `CRYPTO_malloc`

Reply:
I am confused by the additional parameters. I changed it to ```CRYPTO_malloc(sizeof(*timestamp_tm), NULL, 0)``` and it seems to be working. Was that what you were expecting?

Reply:
Oops. sorry. My mistake. I actually meant `OPENSSL_malloc` (which is just a macro for `CRYPTO_malloc`, but without the extra params).

Reply:
Thanks, now I'm not confused anymore. :)